sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/SensorContextUtilsTest.java

```diff
@@ -157,6 +158,90 @@ public void hasFilesOfLanguage_whenOnlyOtherLanguageExists_returnsFalse() {
     assertThat(hasFilesOfLanguage(fs, LANG_ONE)).isFalse();
   }
 
+  @Test
+  public void toTextRange_whenStartAtEOL_doesNotFilterOut() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents(
+        "Some text \n" +
+        "rangeStartingAtEOL1\n" +
+        "Some more text rangeStarting\n" +
+        "AtEOL2 Some other text")
+      .build();
+    fs.add(inputFile);
+
+    var pbTextRange1 = SonarAnalyzer.TextRange.newBuilder()
+      .setStartLine(1)
+      .setStartOffset(10)
+      .setEndLine(2)
+      .setEndOffset(18)
+      .build();
+    assertThat(toTextRange(inputFile, pbTextRange1)).isNotEmpty();
```

Comment:
Now that we have more complex logic for trimming I think it would make sense to check the returned text range values.

Reply:
I have extended the UTs to always check for the resulting range, when the result is not empty.